File: core/src/remove/tanks/game/physics/filter/FilterPrefab.java
package remove.tanks.game.physics.filter;

import com.google.common.base.MoreObjects;
import remove.tanks.game.utility.Prefab;

/**
 * @author Mateusz Długosz
 */
public final class FilterPrefab extends Prefab {
    private final short categoryBit;
    private final short maskBit;

    public FilterPrefab(short categoryBit, short maskBit) {
        this.categoryBit = categoryBit;
        this.maskBit = maskBit;
    }

    public short getCategoryBit() {
        return categoryBit;
    }

    public short getMaskBit() {
        return maskBit;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("categoryBit", categoryBit)
                .add("maskBit", maskBit)
                .toString();
    }
}

File: tools/docs-generator/lib/physics/filter/filter_prefab.py
class FilterPrefab(object):
    def __init__(self, category_bit, mask_bit):
        self.category_bit = str(category_bit)
        self.mask_bit = str(mask_bit)

    def get_category_bit(self):
        return self.category_bit

    def get_mask_bit(self):
        return self.mask_bit

    def __str__(self):
        return "FilterPrefab(category_bit={}, mask_bit={})" \
            .format(self.category_bit, self.mask_bit)
